### PR TITLE
Qualcomm AI Engine Direct - GA Static Granite3.3-2b

### PR DESCRIPTION
**Pull request by chenweng-quic**
### Summary
Add Granite3.3-2b support.

Source model:
image
image

Static llama:
`python examples/qualcomm/oss_scripts/llama/llama.py -b build-android -H mlgtw-linux -s c3b39f15 -m SM8650 --temperature 0 --model_mode kv  --max_seq_len 1024 --prefill_ar_len 128 --decoder_model granite_3_3-2b_instruct --prompt "I would like to learn python, could you teach me with a simple example?" --run_lm_eval --task hellaswag --limit 10 --artifact llama_qnn --kv_updater shift_pointer`

#### Accuracy(hellaswag)(limit=10)
prepare_pt2e: {'acc_norm,none': 0.5}
convert_pt2e: {'acc_norm,none': 0.3}
device: {'acc_norm,none': 0.2}

#### Statistics on SM8650(16a4w_block64)
image

#### Statistics on SM8750(16a4w_block64)
image

### Test plan
`
python backends/qualcomm/tests/test_qnn_delegate.py -k TestExampleLLMScript.test_granite_3_3_2b_instruct --device c3b39f15  --host mlgtw-linux --model SM8650  --build_folder build-android --executorch_root . --artifact_dir ./llama_qnn --llama_artifacts llama_qnn
`

cc @cccclai @cbilgin

### CODE REVIEW

File: examples/models/llama/model_args.py [hunk 1]
```diff
@@ -49,6 +49,7 @@ class ModelArgs:
     model_architecture: str = (
         "LlamaForCausalLM"  # This setting is currently only supported for the QNN backend
     )
+    attention_multiplier: Optional[float] = None
```
Comment (cccclai) [inline examples/models/llama/model_args.py]: Can you add some comments to explain the new arguments for attention_multiplier, logits_scaling and residual_multiplier? I understand we don't have them for the other params, but let's start document them to make it more scalable and readable.

Comment (chenweng-quic) [reply examples/models/llama/model_args.py]: Done, thanks for your suggestion.

**Review by cccclai — APPROVED** [review]
Thank you for the contribution!
